pkg/strategy/dca2/collector.go

```diff
@@ -133,20 +137,28 @@ func (rc *Collector) CollectFinishRounds(ctx context.Context, fromOrderID uint64
 	var round Round
 	for _, order := range orders {
 		// skip not this strategy order
-		if order.GroupID != rc.groupID {
+		if rc.filterGroupID && order.GroupID != rc.groupID {
 			continue
 		}
 
 		switch order.Side {
 		case types.SideTypeBuy:
 			round.OpenPositionOrders = append(round.OpenPositionOrders, order)
 		case types.SideTypeSell:
+			round.TakeProfitOrders = append(round.TakeProfitOrders, order)
+
 			if order.Status != types.OrderStatusFilled {
-				rc.logger.Infof("take-profit order is %s not filled, so this round is not finished. Skip it", order.Status)
+				rc.logger.Infof("take-profit order is %s not filled, so this round is not finished. Keep collecting", order.Status)
 				continue
 			}
 
-			round.TakeProfitOrder = order
+			for _, o := range round.TakeProfitOrders {
+				if types.IsActiveOrder(o) {
+					// Should not happen ! but we only log it
+					rc.logger.Errorf("there is at least one take-profit order (%d) is still active, please check it", o.OrderID)
```

Comment:
you can add prefix "unexpected error"

Reply:
and maybe print the symbol and the order? would prefer using `#{orderID}` format